tests/test_unit/test_load_bboxes.py

```diff
@@ -152,24 +233,25 @@ def assert_dataset(
     )
 
 
-def assert_time_coordinates(ds, fps, start_frame):
+def assert_time_coordinates(ds, fps, start_frame=None, frame_array=None):
     """Assert that the time coordinates are as expected, depending on
-    fps value and start_frame.
+    fps value and start_frame or time_array. start_frame takes precedence
+    over frame_array if both are provided.
     """
     # scale time coordinates with 1/fps if provided
     scale = 1 / fps if fps else 1
 
+    # build frame array if not provided
+    if start_frame is not None:
+        frame_array = np.array(
+            range(start_frame, len(ds.coords["time"].data) + start_frame)
+        )
+    # elif not frame_array:
```

Comment:
should this stay commented out?

Reply:
yes, but I added a value error message to be more clear:

```python
def assert_time_coordinates(ds, fps, start_frame=None, frame_array=None):
    """Assert that the time coordinates are as expected, depending on
    fps value and start_frame or time_array. start_frame takes precedence
    over frame_array if both are provided.
    """
    # scale time coordinates with 1/fps if provided
    scale = 1 / fps if fps else 1

    # build frame array from start_frame if provided
    if start_frame is not None:
        frame_array = np.array(
            range(start_frame, len(ds.coords["time"].data) + start_frame)
        )
    elif frame_array is None:
        raise ValueError(
            "Either start_frame or frame_array must be provided."
            "start_frame takes precedence over frame_array if "
            "both are provided."
        )

    # assert numpy array of time coordinates
    np.testing.assert_allclose(
        ds.coords["time"].data, np.array([f * scale for f in frame_array])
    )
```